Fix dropdown menu button label issue

huijing opened the pull request:
Fixes https://github.com/lorenzolewis/starlight-utils/issues/104

This PR replaces `{Astro.props.starlightProps.labels["menuButton.accessibleLabel"]}` with `{Astro.locals.t("menuButton.accessibleLabel")}`

lorenzolewis commented:
Thank you for this! I believe we will also need to bump the minimum version of Starlight in package.json for the version in which this change occurred. Will you be able to do that as well as add a changeset to this PR? This would be a "minor" change to reflect Starlight's versioning. 

Let me know if you need any help with that and thank you!

huijing commented:
I ran `npx changeset` and added the changeset, however, I'm not sure if I did the package.json bump correctly. The Starlight version with the change was 0.28.0 which requires the Astro version to be 4.14.0 (https://github.com/withastro/starlight/releases/tag/%40astrojs%2Fstarlight%400.28.0), so I changed that too. But again, not sure if that's how it's supposed to work.

lorenzolewis commented on .changeset/heavy-students-exercise.md:
```diff
@@ -0,0 +1,5 @@
+"@lorenzo_lewis/starlight-utils": minor
+---
+
+Update translation string for dropdown menu label with new Starlight localization API
```
```suggestion
Support translated strings introduced in the Starlight 0.28 localization API
```

lorenzolewis commented on packages/starlight-utils/package.json:
```diff
@@ -13,8 +13,8 @@
+    "astro": "^4.14.0"
   },
   "peerDependencies": {
     "@astrojs/starlight": ">=0.16.0",
```
I believe it's the peer dependency that needs updated. The dev dependency is what is used while working on and developing THIS package, whereas the peer dependency lets us give a specific range for someone consuming THIS package. 

I think it's okay to leave the other change you've made here as it needed to be updated anyway 😁

huijing replied:
Ok, I made the bump.

lorenzolewis reviewed (changes requested):
Thanks! A few quick changes and then it should be good to go!

lorenzolewis commented:
Thanks! Just needed to change a few more things with the lock files to make it work with the site used to test.
